packages/typespec-test/test/chatApi_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -40,7 +41,13 @@ export async function _createStreamingDeserialize(
   result: CreateStreaming200Response
 ): Promise<ChatCompletionChunk> {
   if (result.status !== "200") {
-    throw result.body;
+    const internalError = (result.body as any).error || result.body || result;
+    const message = `Unexpected status code ${result.status}`;
+    throw new RestError(internalError.message ?? message, {
+      statusCode: Number(result.status),
+      code: internalError.code,
+      request: result.request,
+    });
```

Comment:
I think we should pull out this logic into a helper function because it is exactly the same. Also, perhaps it should be exported by core.

Reply:
Just found core-client-rest has exported an function called createRestError, which is similar like line 46 to line 49, except createRestError doesn't add the internalError.code into RestError, do you think if we change the above into something like 
```
    const internalError = (result.body as any).error || result.body || result;
    const message = `Unexpected status code ${result.status}`;
    throw createRestError(internalError.message ?? message, result);
```
is enough or we should improve the createRestError ?

Reply:
I propose to update `createRestError` as follows:
- add an overload that takes just the response and take care of creating the message
- set the error code property

This way, the generated code would be just `throw createRestError(result)`. /cc @xirzec

Reply:
looks good